src/app/onboarding/availability/page.tsx

```diff
@@ -1,24 +1,92 @@
 'use client';
 
-import Button from '@/components/Button';
-import COLORS from '@/styles/colors';
+import { useContext, useState, useEffect } from 'react';
 import { H1 } from '@/styles/text';
 import { OnboardingContext } from '@/utils/OnboardingProvider';
-import { useContext } from 'react';
+import TextInput from '@/components/TextInput';
+import { isValidDate } from '@/utils/helpers';
+import DateInput from '@/components/DateInput';
+import { Profile } from '@/types/schema';
+import TextAreaInput from '@/components/TextAreaInput';
 
 export default function Page() {
   const onboarding = useContext(OnboardingContext);
+  const [hours, setHours] = useState('');
+  const [startDate, setStartDate] = useState('');
+  const [periods, setPeriods] = useState('');
+
+  useEffect(() => {
+    setHours(
+      onboarding?.profile.hours_per_month
+        ? onboarding?.profile.hours_per_month.toString()
+        : '',
+    );
+    setStartDate(onboarding?.profile.start_date || '');
+    setPeriods(onboarding?.profile.availability_description || '');
+  }, []);
+
+  const getHoursErrorText = () => {
+    if (hours !== '' && Number.isNaN(parseInt(hours, 10))) {
+      return 'Must be a number';
+    }
+    return '';
+  };
+
+  const getErrorText = () => {
+    if (startDate !== '' && !isValidDate(startDate)) {
+      return 'Must select a current or future date';
+    }
+    return '';
+  };
+
+  useEffect(() => {
+    if (
+      hours !== '' &&
+      !Number.isNaN(parseInt(hours, 10)) &&
+      startDate !== '' &&
+      isValidDate(startDate)
+    ) {
+      // update profile
+      const partialProfile: Partial<Profile> = {
+        hours_per_month: +hours,
+        start_date: startDate,
+        availability_description: periods,
+      };
+      onboarding?.updateProfile(partialProfile);
+      // enable continue
+      onboarding?.setCanContinue(true);
+    } else {
+      onboarding?.setProgress(2);
+      onboarding?.setCanContinue(false);
+    }
+  }, [hours, startDate, periods]);
```

Comment:
I would recommend removing this `useEffect` and use `onChange` event handlers instead to update what is stored in onboarding

Reply:
I would recommend setting `canContinue` in the other `useEffect` as we're populating the data, and use values that's stored in `onboarding` instead of what we have stored as states.